### PR TITLE
WGS: Add joint-discovery workflow.

### PR DESCRIPTION
**Pull request by mbookman**
This pull request adds the joint genotyping workflow used for AMP PD.
The commit includes the original Broad WDL and inputs.json along with the WDL and inputs.json used for AMP PD.

### CODE REVIEW

File: wgs/joint-discovery/README.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+and imported into the FireCloud Methods Repository.
+
+
+A few updates needed to be made to get it running reliably in Terra for AMP PD:
+
+1- Allow for passing a "user_project_id" such that ImportGVCFs can support
+requester pays buckets.
+
+2- Add support for maxRetries, as shards can arbitrarily fail due to
+Pipelines API Error 10
```
Comment (wnojopra) [inline wgs/joint-discovery/README.md]: nit: period after 10

File: wgs/joint-discovery/README.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+Setting the `merge_count` to 3 results in 6,798 shards.
+Without explicitly setting it, the merge_count is 2, resulting in 10,187.
+
+4- Explicitly use a full-priced VM for SNPsVariantRecalibrator.
+This is a single gating node for the next batch of sharded steps.
+Prefer not to use preemptible VMs here.
+
+5- Increase memory available for VariantRecalibrator.
+For runs of around 4,000 samples, the 3g default memory was nowhere near
+sufficient. With higher values, like 32g, this step finished reliably
```
Comment (wnojopra) [inline wgs/joint-discovery/README.md]: nit: uppercase G for 3g and 32g

File: wgs/joint-discovery/joint-discovery-gatk4.wdl [hunk 3]
```diff
@@ -0,0 +1,1077 @@
...
+## LICENSING : 
+## This script is released under the WDL source code license (BSD-3) (see LICENSE in 
+## https://github.com/broadinstitute/wdl). Note however that the programs it calls may 
+## be subject to different licenses. Users are responsible for checking that they are
+## authorized to run all programs before running this script. Please see the docker 
+## page at https://hub.docker.com/r/broadinstitute/genomes-in-the-cloud/ for detailed
+## licensing information pertaining to the included programs.
+
+workflow JointGenotyping {
+  # mbookman: Allow for passing a project ID such that ImportGVCFs
```
Comment (wnojopra) [inline wgs/joint-discovery/joint-discovery-gatk4.wdl:45]: Are you ok with your username being written throughout the file?

File: wgs/joint-discovery/joint-discovery-gatk4.wdl [hunk 4]
```diff
@@ -0,0 +1,1077 @@
...
+                add_interval(w_chr, w_start, w_end)
+            f1.writelines("\n".join(lines_to_write))
+    CODE
+  }
+
+  runtime {
+    memory: "3 GB"
+    maxRetries: max_retries
+    preemptible: preemptible_tries
+    docker: "python:2.7"
```
Comment (wnojopra) [inline wgs/joint-discovery/joint-discovery-gatk4.wdl:1071]: I wonder if python2 images will ever be deprecated. I would guess not any time soon.

**Review by wnojopra — COMMENTED** [review]
Other than a few nits, lgtm. I finely went over the documentation, comments, and parts of the wdl that you modified. I reviewed less finely the rest of the wdl code.

File: wgs/joint-discovery/README.md (hunk 1 above)
Comment (mbookman) [reply wgs/joint-discovery/README.md]: Done.

File: wgs/joint-discovery/README.md (hunk 2 above)
Comment (mbookman) [reply wgs/joint-discovery/README.md]: Updated.

File: wgs/joint-discovery/joint-discovery-gatk4.wdl (hunk 3 above)
Comment (mbookman) [reply wgs/joint-discovery/joint-discovery-gatk4.wdl:45]: My github ID is already part of the public record, so that is fine.

**Review by wnojopra — APPROVED** [review]
LGTM

**Review by ha-duong-technome — APPROVED** [review]

**Pull request merged**